Document new IViewManagerCreateWithProperties and IViewManagerRequiresNativeLayout (#393)
microsoft/react-native-windows-samples: docs/view-managers.md

The `Microsoft.ReactNative.Managed.ReactPackageProvider` is a convenience that makes sure that all native modules and view managers defined within the app project automatically get registered. So if you're creating your view managers directly within the app project, you won't actually want to define a separate `ReactPackageProvider`.

### More extensibility points

> - In some scenarios, a view manager might need to have more context at view creation time in order to decide what kind of control to instantiate.

Author: ```suggestion
- In some scenarios, a view manager might need to have more context at view creation time in order to decide what kind of control to instantiate.
```

> - In some scenarios, a view manager might need to have more context at control creation time in order to decide what kind of control to instantiate.

Reviewer: nit - repetition of 'control' - 'to decide which one/what kind to instantiate

> This is possible by having the view manager implement the `IViewManagerCreateWithProperties` interface. The `CreateWithProperties` method can then access the properties set in JSX by inspecting the `propertyMapReader`.

Author: ```suggestion
This can be achieved by having the view manager implement the `IViewManagerCreateWithProperties` interface. The `CreateWithProperties` method can then access the properties set in JSX by inspecting the `propertyMapReader`.
```

> This is possible by having the view manager implement the `IViewManagerCreateWithProperties` interface. The `CreateWithProperties` method can then access the properties set in JSX by inspecting the `propertyMapReader`.

Reviewer: nit, reword - 'made possible by' | 'achievable by' | 'possible via'

Author: ```suggestion
This can be achieved by having the view manager implement the `IViewManagerCreateWithProperties` interface. The `CreateWithProperties` method can then access the properties set in JSX by inspecting the `propertyMapReader`.
```

}
```

### More extensibility points

> - In some scenarios, a view manager might need to have more context at control creation time in order to decide what kind of control to instantiate.

Author: ```suggestion
- In some scenarios, a view manager might need to have more context at view creation time in order to decide what kind of control to instantiate.
```

> This is possible by having the view manager implement the `IViewManagerCreateWithProperties` interface:

Author: ```suggestion
This can be achieved by having the view manager implement the `IViewManagerCreateWithProperties` interface. 
```
